TypeScript error 2702 with Mapping types when 'generateUnionEnums' is true

Hello!

First of all, thank you very much for this library and the work you do.

I am having a problem when using discriminators and `generateUnionEnums` set to `true`.

Here's an example of what I get:

```ts
export type ExampleCategory = 'CATEGORY_1' | 'CATEGORY_2'

interface BaseExample {
  // ...
  category: ExampleCategory
  // ...
}

type ExampleCategory1 = BaseExample & {
  // ...
}

type BaseExampleCategoryMapping<Key, Type> = {
  category: Key
} & Type

export type Example = BaseExample &
  BaseExampleCategoryMapping<
    ExampleCategory.CATEGORY_1, // error TS2702: 'ExampleCategory' only refers to a type, but is being used as a namespace here.
    ExampleCategory1
  >
```

This behavior is preventing me from migrating to v13, even though issue #564 has already been resolved.

Thanks in advance!

--- a/.changeset/salty-pets-notice.md
+++ b/.changeset/salty-pets-notice.md
@@ -1,0 +1,10 @@
+---
+"swagger-typescript-api": patch
+---
+
+Ensure discriminator mappings use union enum literals.
+
+Resolve discriminator mapping generation to use literal values when
+`generateUnionEnums` is enabled to avoid emitting enum member references.
+Add regression coverage that snapshots the discriminator output with
+union enums.

--- a/src/schema-parser/base-schema-parsers/discriminator.ts
+++ b/src/schema-parser/base-schema-parsers/discriminator.ts
@@ -177,18 +177,22 @@ export class DiscriminatorSchemaParser extends MonoSchemaParser {
       );
     }
 
-    if (
-      mappingPropertySchema?.rawTypeData?.$parsed?.type === SCHEMA_TYPES.ENUM
-    ) {
+    const parsedEnum = mappingPropertySchema?.rawTypeData?.$parsed;
+    if (parsedEnum?.type === SCHEMA_TYPES.ENUM) {
       mappingPropertySchemaEnumKeysMap = lodash.reduce(
-        mappingPropertySchema.rawTypeData.$parsed.enum,
+        parsedEnum.enum,
         (acc, key, index) => {
-          const enumKey =
-            mappingPropertySchema.rawTypeData.$parsed.content[index].key;
-          acc[key] = ts.EnumUsageKey(
-            mappingPropertySchema.rawTypeData.$parsed.typeName,
-            enumKey,
-          );
+          const enumContent = parsedEnum.content?.[index];
+          if (this.config.generateUnionEnums) {
+            const literalValue =
+              enumContent?.value ??
+              (key !== undefined ? ts.StringValue(key) : undefined);
+            if (literalValue !== undefined) {
+              acc[key] = literalValue;
+            }
+          } else if (parsedEnum.typeName && enumContent?.key) {
+            acc[key] = ts.EnumUsageKey(parsedEnum.typeName, enumContent.key);
+          }
           return acc;
         },
         {},

--- a/tests/spec/discriminator/basic.test.ts
+++ b/tests/spec/discriminator/basic.test.ts
@@ -31,4 +31,22 @@ describe("basic", async () => {
 
     expect(content).toMatchSnapshot();
   });
+
+  test("discriminator with union enums", async () => {
+    await generateApi({
+      fileName: "schema",
+      input: path.resolve(import.meta.dirname, "schema.json"),
+      output: tmpdir,
+      silent: true,
+      addReadonly: true,
+      generateClient: false,
+      generateUnionEnums: true,
+    });
+
+    const content = await fs.readFile(path.join(tmpdir, "schema.ts"), {
+      encoding: "utf8",
+    });
+
+    expect(content).toMatchSnapshot();
+  });
 });

